DSpace/dspace-angular
# Issue 815 controlled vocabulary hierarchical facets 7.4

## References
* Fixes #815
* Requires DSpace/DSpace#8600

## Description
Adds support for hierarchic vocabularies for search filters

## Instructions for Reviewers

List of changes in this PR:
* Altered the `SearchHierarchyFilterComponent` to show a 'Browse subject tree' link under the browse facet
* Created a configuration were these facet/vocabulary links are defined
* added the link between subject and srsc vocabulary to the default app configuration
* Altered the tests for this new functionality 

How to test this PR:

1. Make sure you have at least two items, both with a different subject so the subject filter facet shows up in the sidebar.
2. Verify you can see the "Browse subject tree" link at the bottom of the filter
3. Verify clicking it opens the vocabulary configured in angular and you can click any amount of levels deep -> Double check this correctly applies the search filter as expected
4. Verify that levels get seperated by  ` :: `

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
